data_io/data_io.py

```diff
@@ -385,14 +384,91 @@ def __init__(self, *largs, **kwargs):
       raise NotImplemented
 
 
-class RedshiftConnection(BaseConnection):
+class DatabaseConnection(BaseConnection): 
+  """
+    Intermediate connection class with common code and default behavior.
+  """
+
+  def check_spark_session(self, spark):
+    # check for active spark session
+    if spark is None:
+      spark = SparkSession.builder.getOrCreate()
+
+    return spark
+    
+  def get_options(self, default_options, is_read, **kwargs):
+    # We'll set the default options then override with
+    # whatever the user wants
+    options = default_options
+
+    for option, value in kwargs.items():
+      options[option] = value
+
+    # This is clunky, but we needed a way to support query/dbtable
+    if not(isinstance(self, S3Connection)) and is_read and ('query' not in options.keys() and 'dbtable' not in options.keys()):
+      options['query'] = f'SELECT * FROM {self.location.schema}.{self.location.table}'
+    
+    return options
+  
+  def set_options(self, r_w, options):
+    # Set options for the reader object
+    for option, value in options.items():
+      r_w = r_w.option(option, value)
+      
+    return r_w
+  
+  def set_reader(
```

Comment:
`set` and `get` have special meanings in python land - let's reconsider the verb we use. Maybe `build_reader`? Not sure.

Reply:
methods renamed